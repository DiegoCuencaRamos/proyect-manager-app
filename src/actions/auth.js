import { auth, GoogleAuthProvider, FacebookAuthProvider, TwitterAuthProvider, GithubAuthProvider } from "../firebase"

// LOGIN
const login = (uid) => ({
    type: 'auth/login',
    payload: uid
})

// Login-middleware
const startGoogleLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(GoogleAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startFacebookLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(FacebookAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startTwitterLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(TwitterAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startGithubLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(GithubAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startEmailLogin = (email, password, setErrorMessage) => {
    return (dispath, getState) => {
        return auth.signInWithEmailAndPassword(email, password)
            .catch(e => {
                console.log(e.message)
                if(e.message === 'The password is invalid or the user does not have a password.') {
                    setErrorMessage('he password is invalid')
                } else {
                    setErrorMessage(e.message)
                }  
            })
    }
}

const startEmailSignUp = (email, password, setErrorMessage) => {
    return (dispatch, getState) => {
        return auth.createUserWithEmailAndPassword(email, password)
            .catch(e => {
                console.log(e.message)
                setErrorMessage(e.message)
            })
    }
}

// LOGOUT
const logout = () => ({
    type: 'auth/logout'
})

const startLogout = () => {
    return (dipatch, getState) => {
        return auth.signOut()
    }
}

export {
    login,
    startGoogleLogin,
    startFacebookLogin,
    startTwitterLogin,
    startGithubLogin,
    startEmailLogin,
    startEmailSignUp,
    logout,
    startLogout
}