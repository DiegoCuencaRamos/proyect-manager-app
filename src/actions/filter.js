const searchFilterChanged = (searchText = '') => ({
    type: 'filter/serachFilterChanged',
    payload: searchText
})

const sortByFilterChanged = (sortBy) => ({
    type: 'filter/sortByFilterChanged',
    payload: sortBy
})

const visualizeModeFilterChanged = (visualizeMode = 'list') => ({
    type: 'filter/visualizeModeFilterChanged',
    payload: visualizeMode
})

export { searchFilterChanged, sortByFilterChanged, visualizeModeFilterChanged }