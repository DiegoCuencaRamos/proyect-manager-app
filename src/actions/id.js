const setProyectId = id => ({
    type: 'id/setProyectId',
    payload: id
})

const setTaskId = id => ({
    type: 'id/setTaskId',
    payload: id
})

export { setProyectId, setTaskId }