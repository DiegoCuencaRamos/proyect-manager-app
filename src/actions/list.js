const setTextFilter = text => ({
    type: 'list/setTextFilter',
    payload: text
})

const setSortBy = sortBy => ({
    type: 'list/setSortBy',
    payload: sortBy
})

const setCurrentPage = currentPage => ({
    type: 'list/setCurrentPage',
    payload: currentPage
})

const setPageLimit = pageLimit => ({
    type: 'list/setPageLimit',
    payload: pageLimit
})

export {
    setTextFilter,
    setSortBy,
    setCurrentPage,
    setPageLimit
}