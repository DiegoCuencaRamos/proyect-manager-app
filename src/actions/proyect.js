import database from '../firebase'
import moment from 'moment'

// 1. Add proyect
const addProyect = (proyect) => ({
    type: 'proyect/addProyect',
    payload: proyect
})

const startAddProyect = (proyectData = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        const {
            name = '', 
            description = '', 
            status = '', 
            invoice = undefined,
            startDate = moment().startOf('month'),
            endDate = moment().endOf('month')

        } = proyectData
        const proyect = { name, description, status, invoice, startDate, endDate }

        return database.ref(`/users/${uid}/proyects`).push(proyect)
            .then(ref => {
                dispatch(addProyect({
                    id: ref.key,
                    ...proyect
                }))
            })
    }
}

// 2. Edit proyect
const editProyect = (id, updates) => ({
    type: 'proyect/editProyect',
    payload: {
        id,
        updates
    }
})

const startEditProyect = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid

        return database.ref(`/users/${uid}/proyects/${id}`).update(updates)
            .then(() => {
                dispatch(editProyect(id, updates))
            })
    }
}

// 3. Remove proyect
const removeProyect = (id) => ({
    type: 'proyect/removeProyect',
    payload: id
})

const startRemoveProyect = (id) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid

        return database.ref(`/users/${uid}/proyects/${id}`).remove()
            .then(() => {
                dispatch(removeProyect(id))
            })
    }
}

// 4. Set proyect
const setProyects = (proyects) => ({
    type: 'proyect/setProyects',
    payload: proyects
})

const startSetProyects = () => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid

        return database.ref(`/users/${uid}/proyects`).once('value')
            .then(dataSnapshot => {
                const proyects = []

                dataSnapshot.forEach(childDataSnapshot => {
                    proyects.push({
                        id: childDataSnapshot.key,
                        ...childDataSnapshot.val() 
                    })
                })

                dispatch(setProyects(proyects))
            })
    }
}

export { 
    addProyect,
    startAddProyect, 
    editProyect, 
    startEditProyect, 
    removeProyect, 
    startRemoveProyect, 
    setProyects, 
    startSetProyects 
}