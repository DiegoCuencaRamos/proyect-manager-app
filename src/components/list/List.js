import React, { useContext, useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import DashboardContext from '../../contexts/dashboard-context'
import getVisibleProyects from '../../selectors/proyects'
import getVisibleTasks from '../../selectors/tasks'
import ListFilters from './ListFilters'
import ListHeader from './ListHeader'
import ListBody from './ListBody'
import Pagination from '../pagination/Pagination'


const List = () => {
    // Context
    const isProyect = useContext(DashboardContext)
    // Variables
    const items = useSelector(state => state[ isProyect ? 'proyects' : 'tasks' ]) 
    const filters = useSelector(state => state.filters)
    const proyectId = isProyect ? undefined : useSelector(state => state.ids.proyectId)
    const filteredItems = isProyect ? getVisibleProyects(items, filters) : getVisibleTasks(proyectId, items, filters)
    // State 
    const [ currentItems, setCurrentItems ] = useState([])
    const [ pageLimit, setPageLimit ] = useState(5) // Este puede ir en el calendar tambiÃ©n, estoy probando.

    useEffect(() => {
        setCurrentItems(filteredItems)
    }, [items])

    console.log('1: ', filteredItems)

    // Events
    const onItemsChange = (data) => {
        const { currentPage, pageLimit } = data
        const startItem = (currentPage - 1) * pageLimit
        const endItem = currentPage * pageLimit
        const currentItems = filteredItems.slice(startItem, endItem)

        console.log('2: ', filteredItems)

        setCurrentItems(currentItems)
    }

    const onPageLimitChange = (pageLimit) => {
        setPageLimit(pageLimit)
    }

    // Render
    return (
        <div className="list">
            <div className="container">
                <ListFilters />
                <div className="list__table">
                    <ListHeader />
                    <ListBody currentItems={currentItems} />  
                    <Pagination 
                        filteredItems={filteredItems}
                        totalItems={filteredItems.length} 
                        pageLimit={pageLimit}
                        onPageLimitChange={onPageLimitChange}
                        onItemsChange={onItemsChange}
                        pageNeighbours={1}
                    />                  
                </div>
            </div>
        </div>
    )
}

export default List