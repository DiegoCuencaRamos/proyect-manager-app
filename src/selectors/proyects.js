const getFilteredProyects = (proyects, textFilter, sortBy) => {
    return proyects
        // Get filtered proyects by textFilter
        .filter(proyect => proyect.name.toLowerCase().includes(textFilter.toLowerCase()))
        // Order proyects by criteria
        .sort((a, b) => {
            if(sortBy === 'name') {
                return a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1
            }
        })
    
}

export default getFilteredProyects