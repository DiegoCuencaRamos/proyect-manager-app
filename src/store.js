import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store