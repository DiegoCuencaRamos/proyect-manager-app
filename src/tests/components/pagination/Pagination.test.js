import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import store from '../../../store'
import DashboardContext from '../../../contexts/dashboard-context'
import List from '../../../components/list/List'
import { paginationProyects } from '../../fixures/proyects'
import { addProyect } from '../../../actions/proyect'

const renderWithWrapper = () => mount(
    <MemoryRouter initialEntries={[{ key: 'static' }]}>
        <DashboardContext.Provider value={{ isProyect: true }}>
            <Provider store={store}>
                <List />
            </Provider>
        </DashboardContext.Provider>
    </MemoryRouter>
)

// Rendering
test('Should render Pagination correctly with items', () => {
    // Set proyects to the store
    paginationProyects.forEach(proyect => {
        store.dispatch(addProyect(proyect))
    })

    const wrapper = renderWithWrapper()
    expect(wrapper).toMatchSnapshot()
}) 

// Events
describe('Should handle Pagination events', () => {

    const value = 2
    let wrapper

    beforeEach(() => {
        wrapper = renderWithWrapper()
    })

    // PaginationSelect
    describe('Should handle Pagination Select events:', () => {

        test('Should set pagination limit on page limit change', () => {
            wrapper.find('.pagination__limit select').simulate('change', {
                target: { value }
            })
            expect(wrapper.find('.pagination__limit select').prop('value')).toBe(value)
        })
        
        test('Should set current page items on page limit change', () => {        
            wrapper.find('.pagination__limit select').simulate('change', {
                target: { value }
            })
            expect(wrapper.find('.list__body').children().length).toEqual(value)
        })
        
        test('Should set pages on page limit change', () => {        
            wrapper.find('.pagination__limit select').simulate('change', {
                target: { value }
            })
            expect(wrapper.find('PaginationPages').prop('pages')).toEqual([ 1, 2, 'right', 9 ])
        })
    })

    // PaginationPages
    describe('Should handle Pagination Pages events:', () => {

        beforeEach(() => {
            wrapper.find('.pagination__limit select').simulate('change', {
                target: { value }
            })
        })

        test('Should set correct page on page clicked', () => {
            const clickedPage = 2
            wrapper.find(`PaginationPageItem[page=${clickedPage}] a`).simulate('click', { preventDefault: jest.fn() })

            expect(wrapper.find('PaginationPages').prop('currentPage')).toBe(clickedPage)
            expect(wrapper.find('.pagination__active-page').text()).toBe(clickedPage.toString())
            expect(wrapper.find('ListBody').prop('currentItems')).toEqual([ paginationProyects[2], paginationProyects[3] ])
        })

        test('Should set correct page on right page clicked', () => {
            const currentPage = 1
            wrapper.find('[page="right"] a').simulate('click', { preventDefault: jest.fn() })

            expect(wrapper.find('PaginationPages').prop('currentPage')).toBe(currentPage + 3)
            expect(wrapper.find('.pagination__active-page').text()).toBe((currentPage + 3).toString())
            expect(wrapper.find('ListBody').prop('currentItems')).toEqual([ paginationProyects[6], paginationProyects[7] ])
        })
        
        test('Should set correct page on left page clicked', () => {
            const currentPage = 1
            wrapper.find('[page="right"] a').simulate('click', { preventDefault: jest.fn() })
            wrapper.find('[page="left"] a').simulate('click', { preventDefault: jest.fn() })

            expect(wrapper.find('PaginationPages').prop('currentPage')).toBe(currentPage + 3 - 3)
            expect(wrapper.find('.pagination__active-page').text()).toBe((currentPage + 3 - 3).toString())
            expect(wrapper.find('ListBody').prop('currentItems')).toEqual([ paginationProyects[0], paginationProyects[1] ])
        })
    })
})