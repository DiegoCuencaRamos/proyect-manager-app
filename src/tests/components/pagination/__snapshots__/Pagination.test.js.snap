// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Pagination correctly with items 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "static",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "static",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "static",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <List>
        <div
          className="list"
        >
          <div
            className="container"
          >
            <ListFilters>
              <section
                className="filters"
              >
                <div
                  className="container"
                >
                  <div
                    className="filters__content"
                  >
                    <input
                      className="filters__item"
                      onChange={[Function]}
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </section>
            </ListFilters>
            <div
              className="list__table"
            >
              <ListHeader>
                <div
                  className="list__header"
                >
                  <div
                    className="list__header-item"
                  >
                    <div
                      className="list__name"
                      onClick={[Function]}
                    >
                      <span>
                        Name
                      </span>
                      <i
                        className="fas fa-sort-down"
                      />
                    </div>
                    <p
                      className="list__status"
                    >
                      Status
                    </p>
                    <div
                      className="list__invoice"
                      onClick={[Function]}
                    >
                      <span>
                        Invoice
                      </span>
                      <i
                        className="fas fa-sort-down"
                      />
                    </div>
                    <p
                      className="list__dropdown"
                    >
                      Options
                    </p>
                  </div>
                </div>
              </ListHeader>
              <ListBody
                currentItems={
                  Array [
                    Object {
                      "color": "green",
                      "description": "This is my proyect fixure 1",
                      "endDate": 1616968800000,
                      "id": "p1",
                      "invoice": "100",
                      "name": "Proyect 1",
                      "startDate": 1616626800000,
                      "status": "todo",
                    },
                    Object {
                      "color": "red",
                      "description": "This is my proyect fixure 2",
                      "endDate": 1616968800000,
                      "id": "p2",
                      "invoice": "200",
                      "name": "Proyect 2",
                      "startDate": 1616626800000,
                      "status": "doing",
                    },
                    Object {
                      "color": "blue",
                      "description": "This is my proyect fixure 3",
                      "endDate": 1616968800000,
                      "id": "p3",
                      "invoice": "300",
                      "name": "Proyect 3",
                      "startDate": 1616626800000,
                      "status": "done",
                    },
                    Object {
                      "color": "green",
                      "description": "This is my proyect fixure 4",
                      "endDate": 1616968800000,
                      "id": "p4",
                      "invoice": "100",
                      "name": "Proyect 4",
                      "startDate": 1616626800000,
                      "status": "todo",
                    },
                    Object {
                      "color": "red",
                      "description": "This is my proyect fixure 5",
                      "endDate": 1616968800000,
                      "id": "p5",
                      "invoice": "200",
                      "name": "Proyect 5",
                      "startDate": 1616626800000,
                      "status": "doing",
                    },
                  ]
                }
              >
                <div
                  className="list__body"
                >
                  <ListItem
                    color="green"
                    description="This is my proyect fixure 1"
                    endDate={1616968800000}
                    id="p1"
                    invoice="100"
                    key="p1"
                    name="Proyect 1"
                    startDate={1616626800000}
                    status="todo"
                  >
                    <div
                      className="list__body-item"
                      key="p1"
                    >
                      <Link
                        className="list__name"
                        data-id="p1"
                        onClick={[Function]}
                        to="/proyect/p1"
                      >
                        <LinkAnchor
                          className="list__name"
                          data-id="p1"
                          href="/proyect/p1"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            className="list__name"
                            data-id="p1"
                            href="/proyect/p1"
                            onClick={[Function]}
                          >
                            Proyect 1
                          </a>
                        </LinkAnchor>
                      </Link>
                      <p
                        className="list__status list__status--todo"
                      >
                        todo
                      </p>
                      <p
                        className="list__invoice"
                      >
                        100
                      </p>
                      <div
                        className="list__dropdown"
                      >
                        <div
                          className="list__dropdown-button"
                          onMouseEnter={[Function]}
                        >
                          <i
                            className="fas fa-sliders-h"
                          />
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <div
                          className="list__dropdown-content"
                        >
                          <ul>
                            <li
                              className="editBtn"
                              data-id="p1"
                              onClick={[Function]}
                            >
                              Edit
                            </li>
                            <li
                              className="removeBtn"
                              data-id="p1"
                              onClick={[Function]}
                            >
                              Remove
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </ListItem>
                  <ListItem
                    color="red"
                    description="This is my proyect fixure 2"
                    endDate={1616968800000}
                    id="p2"
                    invoice="200"
                    key="p2"
                    name="Proyect 2"
                    startDate={1616626800000}
                    status="doing"
                  >
                    <div
                      className="list__body-item"
                      key="p2"
                    >
                      <Link
                        className="list__name"
                        data-id="p2"
                        onClick={[Function]}
                        to="/proyect/p2"
                      >
                        <LinkAnchor
                          className="list__name"
                          data-id="p2"
                          href="/proyect/p2"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            className="list__name"
                            data-id="p2"
                            href="/proyect/p2"
                            onClick={[Function]}
                          >
                            Proyect 2
                          </a>
                        </LinkAnchor>
                      </Link>
                      <p
                        className="list__status list__status--doing"
                      >
                        doing
                      </p>
                      <p
                        className="list__invoice"
                      >
                        200
                      </p>
                      <div
                        className="list__dropdown"
                      >
                        <div
                          className="list__dropdown-button"
                          onMouseEnter={[Function]}
                        >
                          <i
                            className="fas fa-sliders-h"
                          />
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <div
                          className="list__dropdown-content"
                        >
                          <ul>
                            <li
                              className="editBtn"
                              data-id="p2"
                              onClick={[Function]}
                            >
                              Edit
                            </li>
                            <li
                              className="removeBtn"
                              data-id="p2"
                              onClick={[Function]}
                            >
                              Remove
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </ListItem>
                  <ListItem
                    color="blue"
                    description="This is my proyect fixure 3"
                    endDate={1616968800000}
                    id="p3"
                    invoice="300"
                    key="p3"
                    name="Proyect 3"
                    startDate={1616626800000}
                    status="done"
                  >
                    <div
                      className="list__body-item"
                      key="p3"
                    >
                      <Link
                        className="list__name"
                        data-id="p3"
                        onClick={[Function]}
                        to="/proyect/p3"
                      >
                        <LinkAnchor
                          className="list__name"
                          data-id="p3"
                          href="/proyect/p3"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            className="list__name"
                            data-id="p3"
                            href="/proyect/p3"
                            onClick={[Function]}
                          >
                            Proyect 3
                          </a>
                        </LinkAnchor>
                      </Link>
                      <p
                        className="list__status list__status--done"
                      >
                        done
                      </p>
                      <p
                        className="list__invoice"
                      >
                        300
                      </p>
                      <div
                        className="list__dropdown"
                      >
                        <div
                          className="list__dropdown-button"
                          onMouseEnter={[Function]}
                        >
                          <i
                            className="fas fa-sliders-h"
                          />
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <div
                          className="list__dropdown-content"
                        >
                          <ul>
                            <li
                              className="editBtn"
                              data-id="p3"
                              onClick={[Function]}
                            >
                              Edit
                            </li>
                            <li
                              className="removeBtn"
                              data-id="p3"
                              onClick={[Function]}
                            >
                              Remove
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </ListItem>
                  <ListItem
                    color="green"
                    description="This is my proyect fixure 4"
                    endDate={1616968800000}
                    id="p4"
                    invoice="100"
                    key="p4"
                    name="Proyect 4"
                    startDate={1616626800000}
                    status="todo"
                  >
                    <div
                      className="list__body-item"
                      key="p4"
                    >
                      <Link
                        className="list__name"
                        data-id="p4"
                        onClick={[Function]}
                        to="/proyect/p4"
                      >
                        <LinkAnchor
                          className="list__name"
                          data-id="p4"
                          href="/proyect/p4"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            className="list__name"
                            data-id="p4"
                            href="/proyect/p4"
                            onClick={[Function]}
                          >
                            Proyect 4
                          </a>
                        </LinkAnchor>
                      </Link>
                      <p
                        className="list__status list__status--todo"
                      >
                        todo
                      </p>
                      <p
                        className="list__invoice"
                      >
                        100
                      </p>
                      <div
                        className="list__dropdown"
                      >
                        <div
                          className="list__dropdown-button"
                          onMouseEnter={[Function]}
                        >
                          <i
                            className="fas fa-sliders-h"
                          />
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <div
                          className="list__dropdown-content"
                        >
                          <ul>
                            <li
                              className="editBtn"
                              data-id="p4"
                              onClick={[Function]}
                            >
                              Edit
                            </li>
                            <li
                              className="removeBtn"
                              data-id="p4"
                              onClick={[Function]}
                            >
                              Remove
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </ListItem>
                  <ListItem
                    color="red"
                    description="This is my proyect fixure 5"
                    endDate={1616968800000}
                    id="p5"
                    invoice="200"
                    key="p5"
                    name="Proyect 5"
                    startDate={1616626800000}
                    status="doing"
                  >
                    <div
                      className="list__body-item"
                      key="p5"
                    >
                      <Link
                        className="list__name"
                        data-id="p5"
                        onClick={[Function]}
                        to="/proyect/p5"
                      >
                        <LinkAnchor
                          className="list__name"
                          data-id="p5"
                          href="/proyect/p5"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            className="list__name"
                            data-id="p5"
                            href="/proyect/p5"
                            onClick={[Function]}
                          >
                            Proyect 5
                          </a>
                        </LinkAnchor>
                      </Link>
                      <p
                        className="list__status list__status--doing"
                      >
                        doing
                      </p>
                      <p
                        className="list__invoice"
                      >
                        200
                      </p>
                      <div
                        className="list__dropdown"
                      >
                        <div
                          className="list__dropdown-button"
                          onMouseEnter={[Function]}
                        >
                          <i
                            className="fas fa-sliders-h"
                          />
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <div
                          className="list__dropdown-content"
                        >
                          <ul>
                            <li
                              className="editBtn"
                              data-id="p5"
                              onClick={[Function]}
                            >
                              Edit
                            </li>
                            <li
                              className="removeBtn"
                              data-id="p5"
                              onClick={[Function]}
                            >
                              Remove
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </ListItem>
                </div>
              </ListBody>
              <Pagination
                onItemsChange={[Function]}
                onPageLimitChange={[Function]}
                pageLimit={5}
                pageNeighbours={1}
                totalItems={18}
              >
                <div
                  className="pagination"
                >
                  <PaginationSelect
                    onPageLimitChange={[Function]}
                    pageLimit={5}
                    totalItems={18}
                    totalPages={4}
                  >
                    <div
                      className="pagination__limit"
                    >
                      <select
                        className="filters__item"
                        onChange={[Function]}
                        value={5}
                      >
                        <option
                          value="2"
                        >
                          2 rows
                        </option>
                        <option
                          value="5"
                        >
                          5 rows
                        </option>
                        <option
                          value="10"
                        >
                          10 rows
                        </option>
                        <option
                          value="all"
                        >
                          All rows
                        </option>
                      </select>
                    </div>
                  </PaginationSelect>
                  <PaginationPages
                    currentPage={1}
                    onItemsChange={[Function]}
                    pageLimit={5}
                    pageNeighbours={1}
                    pages={
                      Array [
                        1,
                        2,
                        3,
                        4,
                      ]
                    }
                    setCurrentPage={[Function]}
                    totalPages={4}
                  >
                    <nav
                      className="pagination__pages"
                    >
                      <ul>
                        <PaginationPageItem
                          currentPage={1}
                          goToPage={[Function]}
                          key="1"
                          page={1}
                          pageNeighbours={1}
                          totalPages={4}
                        >
                          <li>
                            <a
                              className="pagination__active-page"
                              onClick={[Function]}
                            >
                              1
                            </a>
                          </li>
                        </PaginationPageItem>
                        <PaginationPageItem
                          currentPage={1}
                          goToPage={[Function]}
                          key="2"
                          page={2}
                          pageNeighbours={1}
                          totalPages={4}
                        >
                          <li>
                            <a
                              className=""
                              onClick={[Function]}
                            >
                              2
                            </a>
                          </li>
                        </PaginationPageItem>
                        <PaginationPageItem
                          currentPage={1}
                          goToPage={[Function]}
                          key="3"
                          page={3}
                          pageNeighbours={1}
                          totalPages={4}
                        >
                          <li>
                            <a
                              className=""
                              onClick={[Function]}
                            >
                              3
                            </a>
                          </li>
                        </PaginationPageItem>
                        <PaginationPageItem
                          currentPage={1}
                          goToPage={[Function]}
                          key="4"
                          page={4}
                          pageNeighbours={1}
                          totalPages={4}
                        >
                          <li>
                            <a
                              className=""
                              onClick={[Function]}
                            >
                              4
                            </a>
                          </li>
                        </PaginationPageItem>
                      </ul>
                    </nav>
                  </PaginationPages>
                </div>
              </Pagination>
            </div>
          </div>
        </div>
      </List>
    </Provider>
  </Router>
</MemoryRouter>
`;
